src/lib/core/data-table/data-table.ts

```diff
@@ -123,12 +157,11 @@ export class CdkTable<T> implements CollectionViewer {
 
     // TODO(andrewseguin): Add trackby function input.
     // Find and construct an iterable differ that can be used to find the diff in an array.
-    this._dataDiffer = this._differs.find([]).create();
+    this._dataDiffer = this._differs.find(this._data).create();
   }
 
   ngOnDestroy() {
-    // TODO(andrewseguin): Disconnect from the data source so
-    //   that it can unsubscribe from its streams.
+    this._onDestroy.next();
```

Comment:
I believe it goes either way, according to [this documentation](http://reactivex.io/documentation/operators/takeuntil.html): `If this second Observable emits an item or sends a termination notification, the Observable returned by TakeUntil stops mirroring the source Observable and terminates.`

Though I do agree that `complete()` looks nicer and more definitive, will change to that.